gcloud/pubsub/_gax.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    from google.gax import CallOptions
+except ImportError:
+    _HAVE_GAX = False
+else:
+    _HAVE_GAX = True
+
+
+    from google.gax.errors import GaxError
+    from google.pubsub.v1.pubsub_pb2 import PubsubMessage
+
+    from gcloud.exceptions import Conflict
+    from gcloud.exceptions import NotFound
+
+
+    class _PublisherAPI(object):
+
+        def __init__(self, gax_api):
+            self._gax_api = gax_api
+
+        def list_topics(self, project):
+            """List topics for the project associated with this API.
+
+            See:
+            https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/list
+
+            :type project: string
+            :param project: project ID
+
+            :rtype: tuple, (list, str)
+            :returns: list of ``Topic`` resource dicts, plus a
+                    "next page token" string:  if not None, indicates that
+                    more topics can be retrieved with another call (pass that
+                    value as ``page_token``).
+            """
+            options = CallOptions(is_page_streaming=False)
+            path = 'projects/%s' % (project,)
+            response = self._gax_api.list_topics(path, options)
+            topics = [{'name': topic_pb.name} for topic_pb in response.topics]
+            return topics, response.next_page_token
+
+        def topic_create(self, topic_path):
+            """API call:  create a topic
+
+            See:
+            https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/create
+
+            :type topic_path: string
+            :param topic_path: fully-qualfied path of the new topic, in format
+                               ``projects/<PROJECT>/topics/<TOPIC_NAME>``.
+
+            :rtype: dict
+            :returns: ``Topic`` resource returned from the API.
+            :raises: :exc:`gcloud.exceptions.Conflict` if the topic already 
+                     exists
+            """
+            try:
+                topic_pb = self._gax_api.create_topic(topic_path)
+            except GaxError:
```

Comment:
@tbetbetbe How should we interpret errors? With gRPC a `LocalError` has an RPC [error code][1] that we can check.

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/67693642144dee7d9c336abfe01885c8b54aae0f/system_tests/bigtable.py#L244-L249